aea/cli/push.py

```diff
@@ -19,76 +19,85 @@
+except ImportError:
+    IS_IPFS_PLUGIN_INSTALLED = False
 
-@push.command(name=CONNECTION)
-@click.argument("connection-id", type=PublicIdParameter(), required=True)
-@pass_ctx
-def connection(ctx: Context, connection_id: PublicId) -> None:
-    """Push a connection to the registry or save it in local registry."""
-    if ctx.config.get("local"):
-        _save_item_locally(ctx, CONNECTION, connection_id)
-    else:
-        push_item(ctx, CONNECTION, connection_id)
 
+@click.command()
+@registry_flag()
+@component_flag(wrap_public_id=False)
+@click.argument("public_id_or_path", type=PublicIdOrPathParameter())
+@click.pass_context
+def push(
+    click_context: click.Context,
+    component_type: str,
+    public_id_or_path: Union[PublicId, Path],
+    registry: str,
+) -> None:
+    """Push a non-vendor package of the agent to the registry."""
 
-@push.command(name=CONTRACT)
-@click.argument("contract-id", type=PublicIdParameter(), required=True)
-@pass_ctx
-def contract(ctx: Context, contract_id: PublicId) -> None:
-    """Push a contract to the registry or save it in local registry."""
-    if ctx.config.get("local"):
-        _save_item_locally(ctx, CONTRACT, contract_id)
+    if isinstance(public_id_or_path, PublicId):
+        push_from_public_id(
+            click_context, component_type, cast(PublicId, public_id_or_path), registry
+        )
     else:
-        push_item(ctx, CONTRACT, contract_id)
+        push_item_from_path(
+            click_context, component_type, cast(Path, public_id_or_path), registry
+        )
 
 
-@push.command(name=PROTOCOL)
-@click.argument("protocol-id", type=PublicIdParameter(), required=True)
-@pass_ctx
-def protocol(ctx: Context, protocol_id: PublicId) -> None:
-    """Push a protocol to the registry or save it in local registry."""
-    if ctx.config.get("local"):
-        _save_item_locally(ctx, PROTOCOL, protocol_id)
+@check_aea_project
+def push_from_public_id(
+    context: click.Context, component_type: str, public_id: PublicId, registry: str
+) -> None:
+    """Push from public id."""
+    ctx = cast(Context, context.obj)
+    if registry == REGISTRY_LOCAL:
+        _save_item_locally(ctx, component_type, public_id)
     else:
-        push_item(ctx, PROTOCOL, protocol_id)
-
+        if get_default_remote_registry() == REMOTE_IPFS:
+            raise click.ClickException(
+                "IPFS registry not supported while pushing an item using public id."
+            )
 
-@push.command(name=SKILL)
-@click.argument("skill-id", type=PublicIdParameter(), required=True)
-@pass_ctx
-def skill(ctx: Context, skill_id: PublicId) -> None:
-    """Push a skill to the registry or save it in local registry."""
-    if ctx.config.get("local"):
-        _save_item_locally(ctx, SKILL, skill_id)
-    else:
-        push_item(ctx, SKILL, skill_id)
+        push_item(ctx, component_type, public_id)
 
 
 def _save_item_locally(ctx: Context, item_type: str, item_id: PublicId) -> None:
```

Comment:
This is no longer used I think? (we have new push_item_local below)

Reply:
It's used when we want to push using a public id instead of package path.